lib/gst/player/gstplayer.c

```diff
@@ -2641,6 +2658,46 @@ gst_player_seek_internal (gpointer user_data)
 }
 
 /**
+ * gst_player_set_rate:
+ * @player: #GstPlayer instance
+ * @rate: playback rate
+ *
+ * Playback at specified rate
+ */
+void
+gst_player_set_rate (GstPlayer *self, gdouble rate)
+{
+  g_return_if_fail (GST_IS_PLAYER (self));
+
+  self->rate = rate;
+  if (rate != 1.0)
+    self->trick_play = TRUE;
+  else
+    self->trick_play = FALSE;
+
+  self->seek_position = gst_player_get_position (self);
+
+  self->seek_source = g_idle_source_new ();
```

Comment:
And maybe also need to do some seek throttling like we do in gst_player_seek() here?